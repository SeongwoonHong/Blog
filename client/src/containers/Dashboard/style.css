#dashboard {
  position: relative; }
  #dashboard .header {
    text-align: center;
    color: #0275d8;
    font-size: 35px; }
  #dashboard .btn-new {
    position: absolute;
    top: 0px;
    right: 30px; }

@media screen and (max-width: 500px) {
  #dashboard .header {
    font-size: 18px; }
  #dashboard .btn-new {
    right: 10px; } }
