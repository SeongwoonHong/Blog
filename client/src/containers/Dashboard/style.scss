#dashboard {
  position: relative;
  .header {
    text-align: center;
    color: #0275d8;
    font-size: 35px;
  }
  .btn-new {
    position: absolute;
    top: 0px;
    right: 30px;
  }
}
@media screen and (max-width: 500px) {
  #dashboard .header {
    font-size: 18px;
  }
  #dashboard .btn-new {
    right: 10px;
  }
}
