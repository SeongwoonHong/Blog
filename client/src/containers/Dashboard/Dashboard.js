import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import TextButton from '../../components/TextButton/TextButton';
import Posts from '../../components/Posts';
import Loader from '../../components/Loader/Loader';
import './style.css';

class Dashboard extends Component {
  componentDidMount = () => {
    if (!this.props.posts.data.length) { // it does not fetch the data if there are post data in redux store already.
      this.props.fetchPostsRequest();
    }
  }
  render() {
    return (
      <div id="dashboard">
        <h3 className="header">Blogs</h3>
        {
          this.props.posts.status === 'WAITING'
          && <Loader />
        }
        {
          this.props.posts.status === 'SUCCESS'
          && <Posts posts={this.props.posts.data} />

        }
        <Link to="/posts/new" className="btn-new">
          <TextButton>
            New
          </TextButton>
        </Link>
      </div>
    );
  }
}
Dashboard.defaultProps = {
  posts: {},
  fetchPostsRequest: () => console.warn('fetchPostsRequest is not emplemented')
};

Dashboard.propTypes = {
  posts: PropTypes.object,
  fetchPostsRequest: PropTypes.func
};
export default Dashboard;
