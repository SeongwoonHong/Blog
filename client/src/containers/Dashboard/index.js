import { connect } from 'react-redux';
import Dashboard from './Dashboard';
import * as actions from '../../actions';

const mapStateToProps = (state) => {
  return {
    posts: state.posts.fetchPosts
  }
}
const mapDispatchToProps = (dispatch) => {
  return {
    fetchPostsRequest: () => {
      return dispatch(actions.fetchPostsRequest());
    }
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
