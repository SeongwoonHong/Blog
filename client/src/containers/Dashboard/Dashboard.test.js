import React from 'react';
import { shallow } from 'enzyme';
import { Dashboard } from './Dashboard';


describe('<Dashboard />', () => {
  let dashboard = shallow(<Dashboard />);
  it('renders properly', () => {
    // console.log(dashboard.debug());
    expect(dashboard).toMatchSnapshot();
  })

  it('contains `Blogs` Header', () => {
    expect(dashboard.find('.header').text()).toEqual('Blogs');
  });

  it('contains <Loader /> when `posts.status` is `WAITING`', () => {
    let props = { posts: { status: 'WAITING' }};
    dashboard = shallow(<Dashboard {...props} />);
    expect(dashboard.find('Loader').exists()).toBe(true);
  });

  it('contains (connected) <Posts /> when `posts.status` is `SUCCESS`', () => {
    let props = { posts: { status: 'SUCCESS' }};
    dashboard = shallow(<Dashboard {...props} />);
    expect(dashboard.find('Connect(Posts)').exists()).toBe(true);
  });

  it('contains a button that has class name `btn-new`', () => {
    // console.log(dashboard.debug());
    expect(dashboard.find('Link.btn-new').exists()).toBe(true);
  });


});
