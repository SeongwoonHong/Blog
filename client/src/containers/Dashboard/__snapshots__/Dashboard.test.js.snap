// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard
    fetchPostsRequest={[Function]}
    posts={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="header"
>
          Blogs
</h3>,
        false,
        false,
        <Link
          className="btn-new"
          replace={false}
          to="/posts/new"
>
          <TextButton
                    className=""
                    id=""
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    role="presentation"
                    style={Object {}}
                    type=""
          >
                    New
          </TextButton>
</Link>,
      ],
      "id": "dashboard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Blogs",
          "className": "header",
        },
        "ref": null,
        "rendered": "Blogs",
        "type": "h3",
      },
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <TextButton
            className=""
            id=""
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="presentation"
            style={Object {}}
            type=""
>
            New
</TextButton>,
          "className": "btn-new",
          "replace": false,
          "to": "/posts/new",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "New",
            "className": "",
            "id": "",
            "onClick": [Function],
            "onMouseEnter": [Function],
            "onMouseLeave": [Function],
            "role": "presentation",
            "style": Object {},
            "type": "",
          },
          "ref": null,
          "rendered": "New",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="header"
>
            Blogs
</h3>,
          false,
          false,
          <Link
            className="btn-new"
            replace={false}
            to="/posts/new"
>
            <TextButton
                        className=""
                        id=""
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        role="presentation"
                        style={Object {}}
                        type=""
            >
                        New
            </TextButton>
</Link>,
        ],
        "id": "dashboard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Blogs",
            "className": "header",
          },
          "ref": null,
          "rendered": "Blogs",
          "type": "h3",
        },
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <TextButton
              className=""
              id=""
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="presentation"
              style={Object {}}
              type=""
>
              New
</TextButton>,
            "className": "btn-new",
            "replace": false,
            "to": "/posts/new",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "New",
              "className": "",
              "id": "",
              "onClick": [Function],
              "onMouseEnter": [Function],
              "onMouseLeave": [Function],
              "role": "presentation",
              "style": Object {},
              "type": "",
            },
            "ref": null,
            "rendered": "New",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
